OsmAnd/src/net/osmand/plus/mapcontextmenu/controllers/AmenityMenuController.java

```diff
@@ -160,10 +160,9 @@ public boolean displayDistanceDirection() {
 	public String getNameStr() {
 		String preferredLang = getPreferredMapLang();
 		if (amenity.getType().isWiki()) {
-			MapActivity mapActivity = getMapActivity();
-			if (mapActivity != null) {
-				OsmandApplication app = mapActivity.getMyApplication();
-				preferredLang = WikipediaPoiMenu.getWikiArticleLanguage(app,
+			WikipediaPlugin wikiPlugin = OsmandPlugin.getPlugin(WikipediaPlugin.class);
```

Comment:
We need to do it via OsmAndPlugin.prepareExtraTopPoiFilters and there specify all what we need for this feature.